src/lib/deploy.js

```diff
@@ -218,22 +217,40 @@ async function deploy({ network, yes }) {
   const { fee } = response;
   if (!fee) return;
 
+  let client = new Client({ network: 'testnet' }); // TODO: Make this configurable for mainnet and testnet.
+  let feePayer = client.derivePublicKey(privateKey); // TODO: Using the zkapp private key to deploy. Should make the 'fee payer' configurable by the user.
+
+  const accountQuery = getAccountQuery(feePayer);
   const graphQLEndpoint = config?.networks[network]?.url ?? DEFAULT_GRAPHQL;
+  let nonce = 0;
+  response = await sendGraphQL(graphQLEndpoint, accountQuery);
 
+  if (response?.data?.account?.nonce) {
+    nonce = response.data.account.nonce;
+  } else {
+    let response = await prompt({
+      type: 'input',
+      name: 'nonce',
+      message: (state) => {
+        const style = state.submitted && !state.cancelled ? green : reset;
+        return style('Please confirm the nonce of the account:');
+      },
+      validate: (val) => {
+        if (!val) return red('Nonce is required.');
+        if (isNaN(val)) return red('Nonce must be a number.');
+        return true;
+      },
+      result: (val) => val.trim().replace(/ /, ''),
+    });
+    nonce = response.nonce;
+  }
   let signedPayment = await step('Sign transaction', async () => {
-    let client = new Client({ network: 'testnet' }); // TODO: Make this configurable for mainnet and testnet.
-    let feePayer = client.derivePublicKey(privateKey); // TODO: Using the zkapp private key to deploy. Should make the 'fee payer' configurable by the user.
-
-    const accountQuery = getAccountQuery(feePayer);
-    const response = await sendGraphQL(graphQLEndpoint, accountQuery);
-
     let feePayerDeploy = {
       feePayer,
+      nonce,
       fee: `${fee}000000000`, // add 9 zeros -- in nanomina (1 billion = 1.0 mina)
```

Comment:
This doesn't work if `fee` is anything with a comma, e.g. `0.1`

I'd recommend this logic instead: `String(Number(fee) * 1e9))`